src/main/java/BMO/task/Deadline.java

```diff
@@ -0,0 +1,57 @@
+package bmo.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+
+    private LocalDate by;
+    private DateTimeFormatter outputFormatter;
+
+    /**
+     * Constructor for Deadline class.
+     *
+     * @param description Description of the task
+     * @param by Deadline date of the task
+     */
+    public Deadline(String description, String by) {
+        super(description);
+
+        //Define input date format
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+
+        // Convert the string to a LocalDate
+        this.by = LocalDate.parse(by, inputFormatter);
+
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM-dd-yyyy");
```

Comment:
As these Formatter will be used by most of the classes inheriting from Task class